subsys/mgmt/mcumgr/lib/cmd/stat_mgmt/src/stat_mgmt.c

```diff
@@ -95,45 +115,69 @@ stat_mgmt_cb_encode(CborEncoder *enc, struct stat_mgmt_entry *entry)
 static int
 stat_mgmt_show(struct mgmt_ctxt *ctxt)
 {
+	struct zcbor_string value;
+	zcbor_state_t *zse = &ctxt->cnbe->zs;
+	zcbor_state_t *zsd = ctxt->cnbd->zs;
 	char stat_name[STAT_MGMT_MAX_NAME_LEN];
-	CborEncoder map_enc;
-	CborError err;
-	int rc = MGMT_ERR_EOK;
-
-	struct cbor_attr_t attrs[] = {
-		{
-			.attribute = "name",
-			.type = CborAttrTextStringType,
-			.addr.string = stat_name,
-			.len = sizeof(stat_name)
-		},
-		{ NULL },
-	};
+	bool ok;
+	size_t counter = 0;
 
-	err = cbor_read_object(&ctxt->it, attrs);
-	if (err != 0) {
-		return MGMT_ERR_EINVAL;
+	if (!zcbor_map_start_decode(zsd)) {
+		return MGMT_ERR_EUNKNOWN;
 	}
 
-	err |= cbor_encode_text_stringz(&ctxt->encoder, "rc");
-	err |= cbor_encode_int(&ctxt->encoder, MGMT_ERR_EOK);
+	/* Only interested in "name" keyword */
+	do {
+		struct zcbor_string key;
+		static const char name_key[] = "name";
+
+		ok = zcbor_tstr_decode(zsd, &key);
+
+		if (ok) {
+			if (key.len == (ARRAY_SIZE(name_key) - 1) &&
+			    memcmp(key.value, name_key, ARRAY_SIZE(name_key) - 1) == 0) {
+				ok = zcbor_tstr_decode(zsd, &value);
+				break;
+			}
 
-	err |= cbor_encode_text_stringz(&ctxt->encoder, "name");
-	err |= cbor_encode_text_stringz(&ctxt->encoder, stat_name);
+			ok = zcbor_any_skip(zsd, NULL);
+		}
+	} while (ok);
+
+	if (!ok || value.len >= ARRAY_SIZE(stat_name)) {
+		return MGMT_ERR_EINVAL;
+	}
 
-	err |= cbor_encode_text_stringz(&ctxt->encoder, "fields");
-	err |= cbor_encoder_create_map(&ctxt->encoder, &map_enc, CborIndefiniteLength);
+	memcpy(stat_name, value.value, value.len);
```

Comment:
As above, value isn't sure to be set if it wasn't included in the cbor map and will have undefined data

Reply:
Fixed

Reply:
The way I see this is: `struct zcbor_string value;` value is undefined, `while()` loop processes all items in message, assuming that the message does contain valid data but no "name" tag and instead is "timber", `ok` will be true but `value` will never be set and will be undefined, so at this point `value.len` and `value.value` could be anything

Reply:
Fixed.